src/app/input-bar/input-bar.component.ts

```diff
@@ -25,15 +25,18 @@ export class InputBarComponent implements OnInit {
 
     // we delete any options that are not filter by metadata keys so that
     // only filter keys are left for the filter pipe to manage.
-    delete options.regex;
     delete options.group;
     this.ccdService.updateFilters(options);
 
   }
 
   parseInput(input: string) {
     try {
-      return JSON.parse(input);
+      if (input) {
+        return JSON.parse(input);
```

Comment:
Should we start implementing our own parser instead of relying on JSON?

Reply:
Yeah sounds like a great next step. I could get started on it if you could give me some kind of run down on the syntax we want to support